docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+Users have requested this functionality as being pivotal to their adoption of ACK (see [#41](https://github.com/aws/aws-controllers-k8s/issues/41)). Information from user feedback has been considered as part of this document.
+
+Below are a number of use cases for requesting the ability this feature.
+
+1. Migrating existing architecture from another platform (CFN, CDK, Terraform, etc.)
+    1. Resources exist but have never been managed by K8s - no spec files
+2. Restoring from a broken K8s environment (etcd failure, complete state loss)
+    1. Resources exist and have previously been managed by K8s, but are not being actively reconciled
+3. Migrating architecture across K8s environments
+    1. Resource exist and are actively being managed by another K8s environment
+
+# Proposed Implementation
+
+## Overview
+
+The proposed solution will provide an alternate workflow by which to adopt resources through a separate set of manifests. A new custom resource definition, the `AdoptedResource` CRD, provides a standardised template by which to indicate to ACK that it should attempt to adopt a given resource. 
+The `AdoptedResource` CRD provides a boilerplate to reference an existing AWS resource (either through ARN, name or other unique identifier) and a target ACK type. Each ACK manager will subscribe to the list of adopted resources and will attempt to adopt any resources that match its managed type.
+The manager will describe the referenced AWS resource, typically through a `Describe*` SDK call, and use the returned values to compile a new custom resource - with filled spec and status fields. The manager will then apply this custom resource to the cluster using the metadata applied in the `AdoptedResource` spec.
+
+## `AdoptedResource` Specification
+
+The `AdoptedResource` custom resource definition will allow for specification of an AWS resource unique identifier, a target ACK type (with API group and kind), and target metadata fields.
+The AWS resource unique identifier allows identification through at most one of:
+
+* `arn` - An [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (Amazon Resource Name)
+* `name` - A resource name (a string value)
+* `id` - A resource ID (an integer value)
+
+The custom resource definition makes no assumptions about which fields should be present for any given target kind. It is therefore up to the manager to validate the identifier input shape for existence and correctness.
+
+The target ACK type is specified through the following required fields:
+
+* [API group and version](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups-and-versioning)
+* [Resource kind](https://kubernetes.io/docs/reference/kubectl/overview/#resource-types)
+
+A validating webhook for the `AdoptedResource` custom resources will validate the existence of the target type in the cluster and ensure the requested type is able to be adopted. 
```

Comment:
Do you envision the webhook to also validate the existence of the AWS resource referred to by the ARN/name/id? Or will the individual ACK service controllers be responsible for that?

Reply:
Absolutely not. I want the validation webhook to be super lightweight, and not care at all about AWS - only about the cluster configuration. Just to provide fast, "apply-time" (idk if that's a concept?) feedback to the user if they're doing something terminally incorrect.